src/main/java/org/opensearch/knn/jni/PlatformUtils.java

```diff
@@ -80,4 +81,38 @@ public static boolean isAVX2SupportedBySystem() {
         }
         return false;
     }
+
+    public static boolean isAVX512SupportedBySystem() {
+
+        if (!Platform.isIntel() || Platform.isMac() || Platform.isWindows()) {
+            return false;
+        }
+
+        if (Platform.isLinux()) {
+            // The "/proc/cpuinfo" is a virtual file which identifies and provides the processor details used
+            // by system. This info contains "flags" for each processor which determines the qualities of that processor
+            // and it's ability to process different instruction sets like mmx, avx, avx2, avx512 and so on.
+            // https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/6/html/deployment_guide/s2-proc-cpuinfo
+            // Here, we are trying to read the details of all processors used by system and find if any of the processor
+            // supports AVX512 instructions supported by faiss.
+            String fileName = "/proc/cpuinfo";
+
+            // AVX512 has multiple flags, which control various features. k-nn requires the same set of flags as faiss to compile
+            // using avx512. Please update these if faiss updates their compilation instructions in the future.
+            // https://github.com/facebookresearch/faiss/blob/main/faiss/CMakeLists.txt
+            String[] avx512 = { "avx512f", "avx512cd", "avx512vl", "avx512dq", "avx512bw" };
+
+            try {
+                return AccessController.doPrivileged((PrivilegedExceptionAction<Boolean>) () -> {
+                    Stream<String> linestream = Files.lines(Paths.get(fileName));
+                    String flags = linestream.filter(line -> line.startsWith("flags")).findFirst().orElse("");
+                    return Arrays.stream(avx512).allMatch(flags::contains);
+                });
+
+            } catch (Exception e) {
+                logger.error("[KNN] Error reading file [{}]. [{}]", fileName, e.getMessage(), e);
```

Comment:
Same as above

Reply:
updated to PrivilegedActionException. 
However, do note that AccessController API itself is deprecated by Java, and hence the IDE prompts the warning. This'll require a broader change across the codebase.